components/bl-totop-component/src/index.js

```diff
@@ -0,0 +1,54 @@
+import { useRef } from "react";
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const Transform = {
+  MAIN: "50%",
+  X: "translateX(-50%)",
+  Y: "translateY(-50%)",
+};
+
+export default function TotopComponent({ component }) {
+  const { classList, offset, position, backgroundColor, color, size, iconSize, indent = "20px", element } = component;
+
+  const styles = {
+    backgroundColor,
+    width: size,
+    height: size,
+    top: position.includes("top") ? indent : position.includes("center-") ? Transform.MAIN : undefined,
+    bottom: position.includes("bottom") ? indent : position.includes("center-") ? Transform.MAIN : undefined,
+    right: position.includes("right") ? indent : position.includes("-center") ? Transform.MAIN : undefined,
+    left: position.includes("left") ? indent : position.includes("-center") ? Transform.MAIN : undefined,
+    transform: position.includes("-center") ? Transform.X : position.includes("center-") ? Transform.Y : undefined,
+  };
+
+  const iconStyles = {
+    width: iconSize,
+    height: iconSize,
+    fill: color,
+  };
+
+  const handleScroll = () => {
+    window.scrollTo({ top: (element?.el?.offsetTop || 0) + offset, behavior: "smooth" });
+  };
+
+  const totopRef = useRef();
+
+  component.el = totopRef.current;
```

Comment:
https://github.com/Backendless/ui-builder-library/pull/195#discussion_r1084203591

>I'm not sure this is the correct way
let's imaging this was rendered only once and the animationOnScrollRef.current before first render will be null and as result it won't work
I assume we need to wrap it with useEffect